### PR TITLE
Add Github Event Collector

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+- display: Use system proxy settings
+  name: proxy
+  type: 8
+  required: false
+- display: Use Secured Connection
+  name: verify
+  defaultvalue: "true"
+  type: 8
+  required: false
+- display: How many logs to fetch
```
Comment (yaakovi) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.yml:49]: follow our standard for max fetch parameters

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        self.client = client
+
+    def _iter_events(self) -> Generator:
+        """
+        Function that responsible for the iteration over the events returned from github api
+        """
+        response = self.client.call()
+        events: list = response.json()
+
+        if len(events) == 0:
```
Comment (yaakovi) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py:99]:
```suggestion
        if not events:
```

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+            return []
+        while True:
```
Comment (yaakovi) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py:101]:
```suggestion

        while True:
```

**Review by yaakovi — APPROVED** [review]
nice

**Review by yaakovi — APPROVED** [review]

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+- display: The maximum number of logs to fetch.
+  name: limit
+  type: 0
+  defaultvalue: 1000
+  required: false
+- display: XSIAM update limit per request
+  name: events_to_add_per_request
+  type: 0
+  defaultvalue: 1000
+  required: false
```
Comment (ShahafBenYakir) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.yml:29]: @adi88d @yaakovi 
Why have we needed this param? Did you encounter issues when sending large amounts of data to XSIAM?

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+            get_events.aggregated_results(limit=1)
+            return_results('ok')
+        elif command == 'github-get-events' or command == 'fetch-events':
+            events = get_events.aggregated_results(limit=int(demisto_params.get('limit')))
+
+            if command == 'fetch-events':
+                if events:
+                    demisto.setLastRun(GetEvents.get_last_run(events))
+                while len(events) > 0:
+                    send_events_to_xsiam(events[:events_to_add_per_request], 'github-audit', 'github-audit')
```
Comment (ShahafBenYakir) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py:176]: @adi88d Vendor should be 'github' and not 'github-audit' in this case.

File: Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+        if command == 'test-module':
+            get_events.aggregated_results(limit=1)
+            return_results('ok')
+        elif command == 'github-get-events' or command == 'fetch-events':
+            events = get_events.aggregated_results(limit=int(demisto_params.get('limit')))
+
+            if command == 'fetch-events':
+                if events:
+                    demisto.setLastRun(GetEvents.get_last_run(events))
+                while len(events) > 0:
```
Comment (ShahafBenYakir) [inline Packs/GitHub/Integrations/GitHubEventCollector/GitHubEventCollector.py:175]: This should be `>=`, the send_events_to_xsiam function also handles empty events case.

**Pull request merged**